api/scpca_portal/loader.py

```diff
@@ -150,13 +149,14 @@ def _create_computed_file(
     Save computed file returned from future to the db.
     Upload file to s3 and clean up output data depending on passed options.
     """
-    # Only upload and clean up projects and the last sample if multiplexed
-    if computed_file.project or computed_file.sample.is_last_multiplexed_sample:
-        if update_s3:
-            s3.upload_output_file(computed_file.s3_key, computed_file.s3_bucket)
-        if clean_up_output_data:
-            computed_file.clean_up_local_computed_file()
+    if update_s3:
+        s3.upload_output_file(computed_file.s3_key, computed_file.s3_bucket)
+    if clean_up_output_data:
+        computed_file.clean_up_local_computed_file()
+
     computed_file.save()
+    if computed_file.sample and computed_file.has_multiplexed_data:
+        ComputedFile.bulk_create_multiplexed_files(computed_file)
```

Comment:
We should enforce a state where all related multiplexed samples either have a computed file or not. We can do that by bulk_creating them instead of saving when the conditions are met.

```suggestion
    if computed_file.sample and computed_file.has_multiplexed_data:
       ComputedFile.bulk_create(
           computed_file.get_multiplexed_computed_files()
       )
    else:
      computed_file.save()
```

```python
def get_multiplexed_computed_files(self):
    computed_files = [self]
    for sample in self.sample.multiplexed_with_samples:
        sample_computed_file = ComputedFile(model_to_dict(computed_file))
        sample.computed_file.pk = None
        sample.computed_file.sample = sample
        computed_files.append(sample_computed_file)
    return computed_files
```